### PR TITLE
Delete hearing day | Error handling

### LINKED ISSUE #8109: Delete hearing day | Error handling
If a user tries to delete a VACOLS hearing day, the server will return an error. We should catch that error on the frontend and display a message that VACOLS hearing days must be deleted from VACOLS.

1) Update deleteHearingDay in DailyDocketContainer to catch errors
1) Add a prop for onErrorHearingDayDelete
1) Display an alert when onErrorHearingDayDelete is true

### PR DESCRIPTION
**Pull request by Sjones352**
Resolves #8109

### Description
This pr shows alert message when you remove  Hearing Day.

### Acceptance Criteria
1) Update deleteHearingDay in DailyDocketContainer to catch errors
1) Add a prop for onErrorHearingDayDelete
1) Display an alert when onErrorHearingDayDelete is true

### Testing Plan
1. Select a regional office without a hearing-day and go to daily docket. 
2. Click Remove Hearing Day button

screen shot 2018-12-07 at 11 30 30 am

### CODE REVIEW

File: client/app/hearingSchedule/reducers.js [hunk 1]
```diff
@@ -283,6 +283,11 @@ const hearingScheduleReducer = (state = initialState, action = {}) => {
         $set: action.payload.date
       }
     });
+  case ACTIONS.HANDLE_DAILY_DOCKET_SERVER_ERROR:
+    return update(state, {
+      dailyDocketServerError: { $set: true }
```
Comment (sharonwarner) [inline client/app/hearingSchedule/reducers.js]: This should also update the displayRemoveHearingDayModal to be null or false.

Comment (Sjones352) [reply client/app/hearingSchedule/reducers.js]: Fixed

**Comment by MeredithStewart** [comment]
The text in the error message looks good! Just want to make sure you knew I reviewed.

File: client/app/hearingSchedule/components/DailyDocket.jsx [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable max-lines */
```
Comment (sharonwarner) [inline client/app/hearingSchedule/components/DailyDocket.jsx]: Do you mind creating a ticket to remove this eslint exception?

**Pull request merged**